src/deps.ts

```diff
@@ -76,6 +75,18 @@ export function getZstdLibrary(): typeof ZStandard | { kModuleError: MongoMissin
   }
 }
 
+/**
```

Comment:
Now that we consume these types in two places, I moved the `AWSCredentials` type into the `deps` module.  You'll also notice that I made the secret and session required.  This aligns with what the Typescript for the AWS SDK says (see the comment I left) and what I see when testing, so I think we can make the type stricter.

An alternative is to leave the properties optional and add logic in the aws credential fetching for FLE that asserts that these required properties are there and errors if they are not.